### PR TITLE
add isBetween() for DateTime

### PR DESCRIPTION
**Pull request by awaik**
Hi Felix,

Made it inclusive, as you mentioned in the issue.

### CODE REVIEW

File: lib/datetime/datetime.dart [hunk 1]
```diff
@@ -102,4 +102,17 @@ extension DateTime_ on DateTime {
...
+
+  /// Returns the [bool] wheather the given date are inside iterval or not.
+  ///
+  /// Example:
+  /// ```dart
+  /// DateTime.isBetween(DateTime(1989, 7, 20), DateTime(1989, 7, 22));
+  /// //the interval is inclusive (now.isBetween(now, now) // true)
+  /// ```
+  bool isBetween(DateTime from, DateTime to) {
+    return this.isAfter(from) && this.isBefore(to) ||
```
Comment (felixblaschke) [inline lib/datetime/datetime.dart]: I would like to see some parentheses here, if we mix AND and OR to show the priority, like
`(EXP1 and EXP2) or EXP3 or EXP4`. It makes it easier to read.

File: test/datetime/datetime_test.dart [hunk 2]
```diff
@@ -95,6 +95,20 @@ void main() {
     var duration = date.duration();
     expect(duration, -85497.hours - 27.minutes - 28.seconds);
   });
+
+  test("datetime isBetween", () {
+    var dateFromOne = DateTime.utc(1959, 3, 31, 14, 32, 32);
+    var dateFromTwo = DateTime.utc(1960, 3, 31, 14, 32, 32);
+    var dateTo = DateTime.utc(1960, 3, 31, 14, 32, 32);
+    expect(true,
```
Comment (felixblaschke) [inline test/datetime/datetime_test.dart]: The correct syntax of the dart's `expect()` method is `expect(actualValue, expectedValue)`.
(https://pub.dev/packages/test)
Just swap the parameters here to match this 😄

**Comment by felixblaschke** [comment]
Hey Alex, It looks a good and simple solution you made here. Good job 👍 . I made some annotations for further polishing.

**Comment by awaik** [comment]
Thank you! Made fixes and made tests more readable.

**Pull request merged**